pkg/arvo/app/dojo.hoon

```diff
@@ -882,13 +883,57 @@
         [[var]~ [%wing var %rep ~]]
       ::
       =/  sam=vase  :(slop ven poz kev)
-      ?.  (~(nest ut p.som) | p.sam)
-        ~>  %slog.1^leaf+"dojo: nest-need"
-        ~>  %slog.0^(skol p.som)
-        ~>  %slog.1^leaf+"dojo: nest-have"
-        ~>  %slog.0^(skol p.sam)
-        !!
-      (slam gat sam)
+      ?:  (~(nest ut p.som) | p.sam)
+        (slam gat sam)
+      ::  something is wrong
+      ::
+      %-  mean
+      ^-  (list tank)
+      =/  cez=type  [%cell %noun [%cell %noun %noun]]
+      ?.  (~(nest ut cez) | p.som)
+        ::  [ven poz kev] can't nest in som
+        ::
+        :~  'dojo: bad gate lost-argument (generator incorrect)'
```

Comment:
Please put the dojo-related error messages below the `nest-need` and `nest-have` messages, since those can be very long and the user would have to scroll up to the see the high-level problem.

Reply:
Fixed, with the exception of keyword args, which are validated outside the primary nest check part of this arm. I don't know if/how it would be possible to make the high level error appear below need / have for this case.